includes/class-indieauth-authorization-endpoint.php

```diff
@@ -238,34 +287,46 @@ public function authorize() {
 			)
 		);
 		$url    = add_query_params_to_url( $args, wp_login_url() );
-		if ( 'code' === $_GET['response_type'] ) {
-			include plugin_dir_path( __DIR__ ) . 'templates/indieauth-authorize-form.php';
-		} elseif ( 'id' === $_GET['response_type'] ) {
+		if ( empty( $scopes ) || empty( array_diff( $scopes, array( 'profile', 'email' ) ) ) || array( 'profile' ) === $scopes ) {
 			include plugin_dir_path( __DIR__ ) . 'templates/indieauth-authenticate-form.php';
+		} else {
+			include plugin_dir_path( __DIR__ ) . 'templates/indieauth-authorize-form.php';
 		}
+
 		include plugin_dir_path( __DIR__ ) . 'templates/indieauth-auth-footer.php';
 	}
 
 	public function confirmed() {
 		$current_user = wp_get_current_user();
+		$user         = $current_user->ID;
 		// phpcs:disable
 		$client_id     = wp_unslash( $_POST['client_id'] ); // WPCS: CSRF OK
 		$redirect_uri  = isset( $_POST['redirect_uri'] ) ? wp_unslash( $_POST['redirect_uri'] ) : null;
 		$scope         = isset( $_POST['scope'] ) ? $_POST['scope'] : array();
 		$code_challenge  = isset( $_POST['code_challenge'] ) ? wp_unslash( $_POST['code_challenge'] ) : null;
 		$code_challenge_method  = isset( $_POST['code_challenge_method'] ) ? wp_unslash( $_POST['code_challenge_method'] ) : null;
+
+		// Do not allow the post scope as deprecated. For compatibility, instead update the offering to the more limited but functionally identical create/update.
```

Comment:
Does the code here never allow me to request a “post” scope? Why? Is this a WordPress-only thing?

Reply:
Because post is deprecated. So it is being updated to create/update.

Reply:
Nitpick: IndieAuth does not deprecate scopes.

I guess this is because Micropub is using something else now and you want to be helpful there? Only nitpicking because something else may want to use IndieAuth as an easy OAuth provider and still want access to post scopes for their purposes. Not really important, just separation of concerns thing.

Reply:
I didn't deprecate it, it just turns it into create update so Micropub doesn't have to support a more open permission.